module TestFailTryDrop {
   err X;

   fun TestFunApply[<Drop> A](consume a:A):() {
       b = a;
       () = try {
          discard(b);
       } catch {
         .X => {
           return;
         };
       };
       return;
   };
}